modules/recon/hosts-hosts/hackertarget_reverse.py

```diff
@@ -7,30 +7,47 @@ class Module(BaseModule):
         'author': 'Qazeer',
         'version': '1.0',
         'description': 'Uses the HackerTarget.com reverse IP lookup API to find host names associated with IP addresses. Updates the \'hosts\' table with the results.',
+        'comments': (
+            'As of October 04 2021, the API limit for the free tier is 20 requests per day.',
+            'If the limit is reached, the API returns "API count exceeded" which raise an error.',
+            'An API key can be specified with "hackertarget_api".',
```

Comment:
If you add the key according to the above comment, it will show up in the module info and there won't be a need for this comment.

Reply:
Problem is, I don't think there is a way to get an API key without subscribing ("[Due to the aggressive nature of some of the security tools, we use the payment as part of the identification process in order to minimize abuse of the system](https://hackertarget.com/scan-membership/)").

Is there a way to specify that the module will work fine without an API key, as it will likely be the majority of use cases, while still supporting the option to use a key?

Reply:
It does this automatically. The framework won't prevent the module from running. For modules that DO require a key at all times, the module just won't work right if the key isn't there, but it doesn't prevent it from running. Keys are not hard requirements for a module to run for this specific reason.

Reply:
I do get that the module will run, its just that I find the warning (`[!] 'hackertarget_api' key not set. hackertarget_reverse module will likely fail at runtime.`) not to be very telling for this case.

Update done in the last commit.

Reply:
Sorry for the late response.

It gives a bad key error:

![image](https://user-images.githubusercontent.com/5614972/144932336-38be46a2-57d1-452b-a5d8-40e684aee81f.png) 

While the same query without specifying an API key correctly returns results (albeit limited):

![image](https://user-images.githubusercontent.com/5614972/144932579-f3300092-6a12-455a-a7ea-fe67929ad70d.png)